### PR TITLE
Update old references and package.json

### LINKED ISSUE #3: Update package.json
Take a pass over `package.json` and update for the new location.

Updates should include, but may not be limited to, the following
- publisher
- repository.url
- bugs.url

### LINKED ISSUE #4: Update references to the old repository
- [nightly.yml](https://github.com/jw3/example-daffodil-vscode/blob/091ad2314af18a81633ef66be996871e20d68a9a/.github/workflows/nightly.yml#L58)
- [README.md](https://github.com/jw3/example-daffodil-vscode/blob/535863991607daf81e5b9372b0b36dbdacdfb21f/README.md#L32)
- [daffodilDebugger.ts#23](https://github.com/jw3/example-daffodil-vscode/blob/8778f5d1403c365419613317eee1a50d980aa1f6/src/daffodilDebugger.ts#L23)
- [daffodilDebugger.ts#48](https://github.com/jw3/example-daffodil-vscode/blob/091ad2314af18a81633ef66be996871e20d68a9a/src/daffodilDebugger.ts#L48)
- [DAPodil.scala](https://github.com/jw3/example-daffodil-vscode/blob/091ad2314af18a81633ef66be996871e20d68a9a/server/core/src/main/scala/org.apache.daffodil.debugger.dap/DAPodil.scala#L754)
- #3

### PR DESCRIPTION
**Pull request by jw3**
Initial updates for all jw3 references and some general cleanup on the `package.json`
- update links and code referring to jw3 repo
- update `package.json` for Apache publishing
- general cleanup on `package.json` based on the example of other ASF VS Code extensions
- removes nightly build

Referenced the following Apache projects with VS Code extensions
- Apache Openwhisk - https://github.com/apache/openwhisk-vscode-extension
- Netbeans LSP server - https://github.com/apache/netbeans/tree/master/java/java.lsp.server/vscode

Also reference the VS Code `package.json` extension manifest docs
- https://code.visualstudio.com/api/references/extension-manifest

closes #3
closes #4

### CODE REVIEW

**Comment by jw3** [comment]
The `package.json` updates here should set #7 up to move forward.

**Review by tuxji — APPROVED** [review]
+1

Found one jw3 left, but it's a comment with an url to an old github issue that probably should stay in that repo.

<https://github.com/apache/daffodil-vscode/blob/091ad2314af18a81633ef66be996871e20d68a9a/server/core/src/main/scala/org.apache.daffodil.debugger.dap/DAPodil.scala#L754>

File: README.md [hunk 1]
```diff
@@ -26,11 +26,11 @@ This is a VS Code extension which enables the interactive debugging of DFDL sche
...
-Until the extension is available in the [VS Code Extension Marketplace](https://marketplace.visualstudio.com/vscode), please download the latest `.vsix` file from the [releases page](https://github.com/jw3/example-daffodil-vscode/releases). Then install it by either:
+Until the extension is available in the [VS Code Extension Marketplace](https://marketplace.visualstudio.com/vscode), please download the latest `.vsix` file from the [releases page](https://github.com/apache/daffodil-vscode/releases). Then install it by either:
   * using the "Extensions: Install from VSIX" command from within VS Code (open the Command Palette with Shift-Command-P, then type `vsix` to bring up the command and pointing it at the downloaded `vsix` file); or
   * on the command-line via `code --install-extension <path-to-downloaded-vsix>`.
 
 
 ## Usage
 
-Please refer to the Wiki page at https://github.com/jw3/example-daffodil-vscode/wiki
+Please refer to the Wiki page at https://github.com/apache/daffodil-vscode/wiki
```
Comment (stevedlawrence) [inline README.md:36]: No where in the README are there build dependencies or instructions. Apache is very source code focused, so instructions should first and foremost be about how to build things from source. It's fine to also provide convenience binaries, whether they be from downloads.apache.org or via the marketplace, but that's importtation information that should be in this README. It doesn't need to be part of this PR, but a new issue should be opened to add that information.

File: package.json [hunk 2]
```diff
@@ -1,13 +1,11 @@
 {
   "name": "daffodil-debugger",
-  "displayName": "Daffodil Debugger",
+  "displayName": "Apache Daffodil Debugger Extension",
+  "description": "VS Code extension for Apache Daffodil Schema Debugger",
```
Comment (stevedlawrence) [inline package.json:4]: Seems `VSCode` without the space is the common way to refer to Visual Studio Code? I would also refer it as ` `DFDL Schema Debugger`, just to make it clear that these are Apache Daffodil schemas, they are DFDL Schemas, and the ebugger is powered by Apache Daffodil.

File: package.json [hunk 3]
```diff
@@ -1,13 +1,11 @@
 {
   "name": "daffodil-debugger",
-  "displayName": "Daffodil Debugger",
+  "displayName": "Apache Daffodil Debugger Extension",
```
Comment (stevedlawrence) [inline package.json:3]: Could the name be changed to make it more clear this is about debugging schemas? With this name, it sounds like it's used for debugging Apache Daffodil, which isn't the main focus of the tool. Maybe something like "DFDL Schema Debugger Extension" or "Apache Daffodil DFDL Schema Debugger Extension", though that's a bit long. Maybe this is fine and the description can be used to clarify it's for schema debugging?

File: package.json [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
+  "description": "VS Code extension for Apache Daffodil Schema Debugger",
   "version": "1.0.0",
-  "publisher": "jw3",
-  "description": "Daffodil Schema Debugger: debug DFDL schema files using Apache Daffodil.",
+  "publisher": "asf",
+  "author": "Apache Daffodil",
   "license": "Apache-2.0",
```
Comment (stevedlawrence) [inline package.json:8]: I usually use `Apache License, Version 2.0`, since this is what https://www.apache.org/licenses/LICENSE-2.0 uses.

**Review by stevedlawrence — APPROVED** [review]
+1 We should update the remaining jw3 reference that @tuxji found to refer to a bug in the new issues tracker. That bug can refer to the old bug if needed.

File: package.json (hunk 2 above)
Comment (jw3) [reply package.json:4]: > Seems VSCode without the space is the common way to refer to Visual Studio Code

You see it both ways in the wild.  I was looking at Openwhisk json when updating this.  Want it changed?

>  would also refer it as DFDL Schema Debugger`

OK

File: package.json (hunk 4 above)
Comment (stevedlawrence) [reply package.json:8]: I wasn't aware of spdx, good to know. Apache-2.0 sounds good then.

File: package.json (hunk 2 above)
Comment (stevedlawrence) [reply package.json]: You're right, the first couple things I saw were VSCode, but I see both uses, even in official vscode documentation. This is fine.

Comment (jw3) [reply package.json]: Yeah I just did a search and we have a mix in this repo, but definitely more with the space.

I added #24 to get this project to a standardization.

**Pull request merged**